corehq/apps/reports/tests/test_generic.py

```diff
@@ -74,3 +80,95 @@ def test_handles_rows(self):
         self.assertEqual(result[0], ['One', 'Two'])
         self.assertEqual(result[1], ['Three', 'Four'])
         self.assertEqual(result[2], ['Five', 'Six'])
+
+
+class GetSerializableReportParameters(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.user = WebUser(domain='test-domain', username='test-user')
+        cls.user.save()
+        cls.addClassCleanup(cls.user.delete, deleted_by_domain='test-domain', deleted_by=None)
+        cls.report = object.__new__(get_report_class('daily_form_stats'))
+        GET_params = QueryDict('', mutable=True)
+        GET_params.update({
+            'emw': ['t__0', 't__5'],
+            'sub_time': '',
+            'startdate': '2022-03-20',
+            'enddate': '2022-03-28'
+        })
+        cls.report_params = {
+            'request': {
+                'GET': GET_params,
+                'META': {
+                    'QUERY_STRING': '',
+                    'PATH_INFO': '/a/practice-tracker/reports/export/daily_form_stats/'
+                },
+                'couch_user': cls.user._id,
+                'can_access_all_locations': True
+            },
+            'request_params': {
+                'sub_time': '',
+                'startdate': '2022-03-20T00:00:00',
+                'enddate': '2022-03-28T00:00:00'
+            },
+            'domain': 'practice-tracker',
+            'context': {}
+        }
+        cls.report.set_report_parameters(cls.report_params)
+
+    def test_returns_serializable_state(self):
+        serializable_state = self.report.get_json_report_parameters()
+        try:
+            json.dumps(serializable_state)
+        except TypeError:
+            self.fail('Expected json serializable state.')
```

Comment:
This seems to overlap with `test_query_dict_elements_preserved_after_serializing`. While I am recommending that that test be moved to a more focused module, if this test is supposed to alert us when serialization breaks, I think it makes sense to actually verify the full serialization. Within the test, make it obvious what state the report has, and then verify that the serialized version matches the expected state. That proves that serialization doesn't just produce output, but produces the correct output.